metaflow/plugins/cards/card_cli.py

```diff
@@ -56,58 +113,50 @@ def resolve_card(
         pathspec: pathspec
         hash (optional): This is to specifically resolve the card via the hash. This is useful when there may be many card with same id or type for a pathspec.
         type : type of card
+        card_id : `id` given to card
+        no_echo : if set to `True` then supress logs about pathspec resolution.
     Raises:
         CardNotPresentException: No card could be found for the pathspec
 
     Returns:
         (card_paths, card_datastore, taskpathspec) : Tuple[List[str], CardDatastore, str]
     """
-    if len(pathspec.split("/")) != 3:
-        raise CommandException(
-            msg="Expecting pathspec of form <runid>/<stepname>/<taskid>"
-        )
-
     flow_name = ctx.obj.flow.name
-    run_id, step_name, task_id = None, None, None
-    # what should be the args we expose
-    run_id, step_name, task_id = pathspec.split("/")
-    pathspec = "/".join([flow_name, pathspec])
-    # we set namespace to be none to avoid namespace mismatch error.
-    namespace(None)
-    task = Task(pathspec)
-    print_str = "Resolving card: %s" % pathspec
+    (
+        task,
+        run_id,
+        resolved_from,
+    ) = resolve_task_from_pathspec(flow_name, pathspec)
```

Comment:
`run_id` and `resolved_from` do not seem to be used except to raise an exception. The logic in `resolve_task_from_pathspec` is fairly long and could probably be simplified if the exception could be simplified a bit. Further, based on the logic in that function, there are cases where when task is none, run_id is also none and doesn't contain a lot of useful information. Perhaps we can simplify and just say that the string given as pathspec does not allow us to resolve to a task. You could, if you wanted to, also give a warning about resolving to multiple tasks.

Reply:
I can refactor the function by raising the error from inside `resolve_task_from_pathspec`and also returning only a `pathspec`.